src/anomalib/utils/metrics/perimg/common.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+import torch
+from torch import Tensor
+
+# =========================================== ARGS VALIDATION ===========================================
+
+
+def _validate_rate_curve(curve: Tensor, nan_allowed: bool = False):
+    if not isinstance(curve, Tensor):
+        raise ValueError(f"Expected argument `curve` to be a Tensor, but got {type(curve)}.")
+
+    if curve.ndim != 1:
+        raise ValueError(f"Expected argument `curve` to be a 1D tensor, but got {curve.ndim}D tensor.")
+
+    if not torch.is_floating_point(curve):
+        raise ValueError(f"Expected argument `curve` to have dtype float, but got {curve.dtype}.")
+
+    if not nan_allowed:
+        if torch.isnan(curve).any():
+            raise ValueError("Expected argument `curve` to not contain NaN values, but got NaN values.")
+        valid_values = curve
+    else:
+        valid_values = curve[~torch.isnan(curve)]
+
+    if (valid_values < 0).any() or (valid_values > 1).any():
+        raise ValueError(
+            "Expected argument `curve` to have values in the interval [0, 1], but got values outside this interval."
+        )
+
+    diffs = curve.diff()
+    diffs_valid = diffs if not nan_allowed else diffs[~torch.isnan(diffs)]
+
+    if (diffs_valid > 0).any():
+        raise ValueError(
+            "Expected argument `curve` to be monotonically decreasing, but got non-monotonically decreasing values."
+        )
+
+
+def _validate_perimg_rate_curves(curves: Tensor, nan_allowed: bool = False):
```

Comment:
matching snake_case convention
```suggestion
def _validate_per_img_rate_curves(curves: Tensor, nan_allowed: bool = False):
```

Reply:
refactor branch